src/TypeScript/Analyzer/JavaScriptAnalyzer.cs

```diff
@@ -113,6 +121,20 @@ internal async Task ExecuteAnalysis(string filePath, IIssueConsumer consumer, Ca
             }
         }
 
+        private async Task<IEslintBridgeClient> GetJavaScriptClient()
+        {
+            var port = await eslintBridgeProcess.Start();
+
+            if (port != javascriptServerPort)
+            {
+                javascriptServerPort = port;
+                javascriptClient?.Dispose();
```

Comment:
There is a problem at the moment with the [EslintBridgeClient.Dispose implementation](https://github.com/SonarSource/sonarlint-visualstudio/blob/branch-replacejs/src/TypeScript/EslintBridgeClient/EslintBridgeClient.cs#L95) -- it attempts to call the "Close" endpoint, but if the process died, the client will fail and not dispose the httpWrapper. I think I should either remove the Close from the dispose, or wrap it with try-catch. Leaning towards removing the Close, WDYT?

Reply:
Hhmmm. What is going to responsible for telling the eslintbridge to close down cleanly? Something should - we can't just terminate the process.
To put it another way, what is responsible for managing the lifetime of the eslintbridge?

I think there is still something missing, but I'm not sure quite what.

Reply:
At the moment, nothing. We discussed that we would want to terminate the process on solution close, but it's not implement yet so nothing calls Close except Dispose.